xmake/modules/core/tools/clang.lua

```diff
@@ -267,8 +280,23 @@ function nf_runtime(self, runtime, opt)
                     llvm_rootdir = _get_llvm_rootdir(self)
                 end
                 if llvm_rootdir then
-                    maps["c++_static"] = table.join(maps["c++_static"], "-L" .. path.join(llvm_rootdir, "lib"))
-                    maps["c++_shared"] = table.join(maps["c++_shared"], "-L" .. path.join(llvm_rootdir, "lib"))
+                    local libdir = path.absolute(path.join(llvm_rootdir, "lib"))
+                    maps["c++_static"] = table.join(maps["c++_static"], "-L" .. libdir)
+                    maps["c++_shared"] = table.join(maps["c++_shared"], "-L" .. libdir)
+                    -- sometimes llvm runtimes are located in a target-triple subfolder
+                    local target_triple = _get_llvm_target_triple(self)
+                    local triple_libdir = (target_triple and os.isdir(path.join(libdir, target_triple))) and path.join(libdir, target_triple)
+                    if triple_libdir then
+                        maps["c++_static"] = table.join(maps["c++_static"], "-L" .. triple_libdir)
+                        maps["c++_shared"] = table.join(maps["c++_shared"], "-L" .. triple_libdir)
+                    end
+                    -- add rpath to avoid the user need to set LD_LIBRARY_PATH by hand
+                    if not self:is_plat("windows", "mingw") then
+                        maps["c++_shared"] = table.join(maps["c++_shared"], "-Wl,-rpath=" .. libdir)
+                        if triple_libdir then
+                            maps["c++_shared"] = table.join(maps["c++_shared"], "-Wl,-rpath=" .. triple_libdir)
```

Comment:
Has it been tested on macos? it should use -Xlinker -rpath -Xlinker path and `-install_name`.

https://github.com/xmake-io/xmake/blob/2c67c028dffe3e456e8b9ac16296ef042629b88b/xmake/modules/core/tools/gcc.lua#L358

https://github.com/xmake-io/xmake/blob/2c67c028dffe3e456e8b9ac16296ef042629b88b/xmake/modules/core/tools/gcc.lua#L499

I think we should use `toolchain:add("runenvs", ..)`to add LD_LIBRARY_PATH/DYLD_LIBRARY_PATH in clang/llvm toolchains. It will be better.

Reply:
it has been tested yes, but i'll try with your suggestions

Reply:
-install_name isn't automatically handled by the gcc inheritance ?